conda/core/solve.py

```diff
@@ -642,8 +645,8 @@ def _add_specs(self, ssc):
         elif ssc.update_modifier == UpdateModifier.UPDATE_SPECS:
             for spec in self.specs_to_add:
                 spec = MatchSpec(spec)
-                if (spec.name in pin_overrides and not ssc.ignore_pinned
-                        or spec.name in ssc.specs_from_history_map):
```

Comment:
I'm removing the specs_from_history_map thing here because the specs are explicitly being updated here.  They should not be held back.